src/CBufParser.cpp

```diff
@@ -796,6 +796,7 @@ bool convert_element_string(const ast_element* elem, const u8*& bin_buffer, size
     }
     if (!dst_elem->is_dynamic_array) {
       check_dst_array = true;
+      dst_array_size = dst_elem->array_suffix->size;
```

Comment:
I was seeing a test failure without this (`names` array of `complex_thing` struct had five empty string entries after decoding instead of the five serialized strings). I don't know why this test failures was not showing up previously. @verdantrobotics?

Reply:
I traced it down to an error in my previous PR: <https://github.com/Verdant-Robotics/cbuf/pull/3/files#r1323495620>

It was not caught by unit testing because pulling in the vlog library forcefully changed the CMake configuration to "RelWithDebInfo", which disabled the `assert()` statements our unit tests rely on for test failures.

Reply:
Looks good, we have to get the size from the metadata if the array is not dynamic.